Mathlib/Analysis/SpecialFunctions/Arcosh.lean

```diff
@@ -115,4 +127,63 @@ def coshPartialEquiv : PartialEquiv ℝ ℝ where
   left_inv' _ hr := arcosh_cosh hr
   right_inv' _ hr := cosh_arcosh hr
 
+theorem continuousOn_arcosh : ContinuousOn arcosh (Ici 1) :=
+  have {x : ℝ} (hx : x ∈ Ici 1) : 0 < x + √(x ^ 2 - 1) :=
+    add_pos_of_pos_of_nonneg (show 0 < x by grind) (sqrt_nonneg _)
+  continuousOn_log.comp (Continuous.continuousOn (by continuity)) (fun _ _ => by grind)
+
+/-- `Real.cosh` as an `OpenPartialHomemorph`. -/
+def coshOpenPartialHomeomorph : OpenPartialHomeomorph ℝ ℝ where
+  toFun := cosh
+  invFun := arcosh
+  source := Ioi 0
+  target := Ioi 1
+  map_source' _ hr := one_lt_cosh.mpr (ne_of_lt hr).symm
+  map_target' _ hr := arcosh_pos hr
+  left_inv' _ hr := arcosh_cosh (le_of_lt hr)
+  right_inv' _ hr := cosh_arcosh (le_of_lt hr)
+  open_source := isOpen_Ioi
+  open_target := isOpen_Ioi
+  continuousOn_toFun := continuous_cosh.continuousOn
+  continuousOn_invFun := continuousOn_arcosh.mono Ioi_subset_Ici_self
+
+theorem hasStrictDerivAt_arcosh {x : ℝ} (hx : x ∈ Ioi 1) :
+    HasStrictDerivAt arcosh (√(x ^ 2 - 1))⁻¹ x := by
+  convert coshOpenPartialHomeomorph.hasStrictDerivAt_symm hx ?_
+    (hasStrictDerivAt_cosh _) using 2
+  · exact (sinh_arcosh (le_of_lt hx)).symm
+  · rw [ne_eq, sinh_eq_zero, coshOpenPartialHomeomorph, OpenPartialHomeomorph.mk_coe_symm,
+      PartialEquiv.coe_symm_mk]
+    exact ne_of_gt (arcosh_pos hx)
+
+theorem hasDerivAt_arcosh {x : ℝ} (hx : x ∈ Ioi 1) : HasDerivAt arcosh (√(x ^ 2 - 1))⁻¹ x :=
+  (hasStrictDerivAt_arcosh hx).hasDerivAt
+
+theorem differentiableAt {x : ℝ} (hx : x ∈ Ioi 1) : DifferentiableAt ℝ arcosh x :=
+  (hasDerivAt_arcosh hx).differentiableAt
+
+theorem differentiableOn_arcosh : DifferentiableOn ℝ arcosh (Ioi 1) := fun _ hx =>
+  (differentiableAt hx).differentiableWithinAt
+
+theorem contDiffAt_arcosh {n : ℕ∞} {x : ℝ} (hx : x ∈ Ioi 1) : ContDiffAt ℝ n arcosh x := by
+  refine coshOpenPartialHomeomorph.contDiffAt_symm_deriv ?_ hx (hasDerivAt_cosh _)
+    contDiff_cosh.contDiffAt
+  rw [ne_eq, sinh_eq_zero]
+  exact ne_of_gt (arcosh_pos hx)
+
+theorem contDiffOn_arcosh {n : ℕ∞} : ContDiffOn ℝ n arcosh (Ioi 1) := fun _ hx =>
```

Comment:
Is it possible to get `AnalyticOn`? We have it for `cosh` ([Real.analyticOn_cosh](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Analysis/SpecialFunctions/Trigonometric/DerivHyp.html#Real.analyticOn_cosh))